File: ProjectEuler/Problems/Problem010.cs
using Microsoft.Extensions.Logging;

namespace ProjectEuler.Problems
{
    public class Problem010 : Problem<long>
    {
        public Problem010(ILogger<Problem010> logger) : base(logger) { }

        protected override long CalculateAnswer()
        {
            var answer = 0L;
            var n = 2000000L;
            var compositeSieve = MathHelpers.IndexCompositeBySieve(n);

            for (var i = 2L; i < n; ++i)
            {
                if (!compositeSieve[i]) answer += i;
            }

            return answer;
        }
    }
}

File: ProjectEulerPython/problems/problem_010.py
import math_helpers
from problems.problem import Problem

class Problem010(Problem):

  def calculate_answer(self) -> int:
    answer = 0
    n = 2000000
    prime_sieve = math_helpers.index_prime_by_sieve(n)

    for i in range(2, n):
      if prime_sieve[i]:
        answer += i

    return answer
